crates/polars-plan/src/plans/aexpr/schema.rs

```diff
@@ -3,7 +3,8 @@ use recursive::recursive;
 use super::*;
 
 fn float_type(field: &mut Field) {
-    if (field.dtype.is_numeric() || field.dtype == DataType::Boolean)
+    if (field.dtype.is_numeric()
+        || matches!(field.dtype, DataType::Decimal(..) | DataType::Boolean))
```

Comment:
Datatype variant is feature flagged.

Though I think we should remove that flag and have all dtypes, but only the implementations feature flagged.

Reply:
@ritchie46 It's a no-win situation, because then you need to feature flag calls to those implementations =/

Reply:
Alright, then feature gate the datatype. :shrug: